crates/sui-indexer/src/db.rs

```diff
@@ -122,16 +136,67 @@ pub fn reset_database(conn: &mut PoolConnection) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+/// Checks that the local migration scripts is a prefix of the records in the database.
+/// This allows us run migration scripts against a DB at anytime, without worrying about
+/// existing readers fail over.
+/// We do however need to make sure that whenever we are deploying a new version of either reader or writer,
+/// we must first run migration scripts to ensure that there is not more local scripts than in the DB record.
+pub fn check_db_migration_consistency(conn: &mut PoolConnection) -> Result<(), IndexerError> {
```

Comment:
iiuc the prefix check is about, a pod can continue working when the binary local schema is behind the DB applied schema, is it expected in our release cycle? assuming that we will deploy indexer writer, reader and graphql following the same cadence.